File: csharp/Units/LengthUnits.cs
namespace UnitConverter
{
    /// <summary>
    /// Example of a metric length unit type in metres which accepts a scalar modifier such as kilo
    /// </summary>
    public sealed class Metre : ILengthUnit
    {
        public static ILengthUnit Create(INumericUnitModifier modifier)
        {
            if (modifier is null) throw new ArgumentNullException($"{nameof(modifier)} cannot be null");
            return new Metre(modifier);
        }

        private Metre(INumericUnitModifier modifier)
        {
            _modifier = modifier;
        }

        /// <summary>
        /// Scalar modifier allowed for Metre
        /// </summary>
        private readonly INumericUnitModifier _modifier;

        public string Name => $"{_modifier.Name}{nameof(Metre)}";

        public string Symbol => $"{_modifier.Symbol}m";

        /// <summary>
        /// Conversion to metre accounting for the modifier
        /// </summary>
        public double ToBase(double value) => value*_modifier.ToBase(1);

        /// <summary>
        /// Conversion from metre accounting for the modifier
        /// </summary>
        public double FromBase(double value) => value*_modifier.FromBase(1);
    }

    /// <summary>
    /// Example of an imperical length unit type in feet
    /// </summary>
    public sealed class Foot : ILengthUnit
    {
        public static ILengthUnit Create()
        {
            return new Foot();
        }

        private Foot() {}

        public string Name => $"{nameof(Foot)}";

        public string Symbol => $"ft";

        /// <summary>
        /// Conversion to metre
        /// </summary>
        public double ToBase(double value) => 0.3048*value;

        /// <summary>
        /// Conversion from metre
        /// </summary>
        public double FromBase(double value) => value/0.3048;
    }
}

File: python/Units/LengthUnits.py
from Model.Interfaces import NumericUnitModifier
from Model.NumericUnitTypes import LengthUnit
from Unitless.Scalars import Unitary

"""
Definition of unit 'Metre'
"""
class Metre(LengthUnit):
    def __init__(self, m=None):
        if m is None:
            m = Unitary()
        if not isinstance(m, NumericUnitModifier):
            raise TypeError(f"Cannot specify {type(m)} as the scalar modifier of {Metre}")
        self._mod = m

    @property
    def Name(self) -> str:
        return f"{self._mod.Name}Metre"

    @property
    def Symbol(self) -> str:
        return f"{self._mod.Symbol}m"

    def ToBase(self, v) -> float:
        return v * self._mod.ToBase(1)

    def FromBase(self, v) -> float:
        return v * self._mod.FromBase(1)

"""
Definition of unit 'Foot'
"""    
class Foot(LengthUnit):
    def __init__(self, m=None):
        if m is None:
            m = Unitary()
        if not isinstance(m, NumericUnitModifier):
            raise TypeError(f"Cannot specify {type(m)} as the scalar modifier of {Foot}")
        self._mod = m

    @property
    def Name(self) -> str:
        return f"{self._mod.Name}Foot"

    @property
    def Symbol(self) -> str:
        return f"{self._mod.Symbol}ft"

    def ToBase(self, v) -> float:
        return v * 0.3048 * self._mod.ToBase(1)

    def FromBase(self, v) -> float:
        return v / 0.3048 * self._mod.FromBase(1)
